src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1909,12 +1910,23 @@ bool KeyCondition::extractMonotonicFunctionsChainFromKey(
 
                     if (date_time_parsing_functions.contains(func_name))
                     {
-                        const auto & arg_types = func_base->getArgumentTypes();
-                        if (!arg_types.empty() && isStringOrFixedString(arg_types[0]))
-                            func_name = func_name + "OrNull";
-                    }
+                        // In case when session timezone is used while querying, skipping the analysis
+                        if (!context->getSettingsRef()[Setting::session_timezone].value.empty())
```

Comment:
Because of this check some tests in `03173_set_transformed_partition_pruning.sql` will fail (note, that you will not see this in fasttest because it does not randomize `session_timezone`)